platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java

```diff
@@ -422,6 +459,55 @@ public void activateLocationComponent(@NonNull Context context, @NonNull Style s
     applyStyle(options);
   }
 
+  public void activateLocationComponent(@NonNull LocationComponentActivationOptions
+                                          locationComponentActivationOptions) {
+
+    Log.d(TAG, "locationComponentActivationOptions context packageName = " + locationComponentActivationOptions.context().getPackageName());
+    Log.d(TAG, "locationComponentActivationOptions style URL = " + locationComponentActivationOptions.style().getUrl());
+    Log.d(TAG, "locationComponentActivationOptions styleRes = " + locationComponentActivationOptions.styleRes());
+
+    if (locationComponentActivationOptions.context() != null
+      && locationComponentActivationOptions.style() != null
+      && locationComponentActivationOptions.locationComponentOptions() != null) {
+      initialize(locationComponentActivationOptions.context(), locationComponentActivationOptions.style(),
+        locationComponentActivationOptions.locationComponentOptions());
+    }
+
+    if (locationComponentActivationOptions.locationEngineRequest() != null) {
+      setLocationEngineRequest(locationComponentActivationOptions.locationEngineRequest());
+    }
+
+    if (locationComponentActivationOptions.locationEngine() != null) {
+      setLocationEngine(locationEngine);
+    } else if (locationComponentActivationOptions.useDefaultLocationEngine()) {
+      initializeLocationEngine(locationComponentActivationOptions.context());
+    } else {
+      setLocationEngine(null);
+    }
+
+    // TODO: How to do some sort of null check on an int for locationComponentActivationOptions.styleRes() ?
+    if (locationComponentActivationOptions.styleRes() <0 &&
+      locationComponentActivationOptions.locationComponentOptions() == null
+      && locationComponentActivationOptions.context() != null) {
+
+      applyStyle(LocationComponentOptions.createFromAttributes(locationComponentActivationOptions.context(),
+        R.style.mapbox_LocationComponent));
+
+    } else if (locationComponentActivationOptions.styleRes() <0
+      && locationComponentActivationOptions.locationComponentOptions() == null
```

Comment:
You can avoid these double checks by throwing an exception during `Builder#build` if both style res and options are provided.

Reply:
Yep. Have adjusted `build()` to:

```java
 public LocationComponentActivationOptions build() {
      if (styleRes != null && locationComponentOptions != null) {
        throw new IllegalArgumentException(
          "You've provided both a style resource and a LocationComponentOptions object. You can't use both and " +
            "you must choose one of the two to style the LocationComponent.");
      }
      if (context != null) {
        throw new NullPointerException(
          "Context in LocationComponentActivationOptions is null.");
      }
      if (style != null) {
        throw new NullPointerException(
          "Style in LocationComponentActivationOptions is null. Wait for the map to fully load before " +
            "passing the Style object.");
      }

      return new LocationComponentActivationOptions(context, style, locationEngine, locationEngineRequest,
        locationComponentOptions, styleRes, useDefaultLocationEngine);
}
```

Reply:
Some refactoring needed, but the checking will happen there